brainglobe_utils/brainmapper/transform_widget.py

```diff
@@ -0,0 +1,780 @@
+        """
+        layers_in_place = self.check_layers()
+        if not layers_in_place:
+            return
+
+        self.status_label.setText("Loading brainreg data ...")
+        data_loaded = self.load_brainreg_directory()
+
+        if not data_loaded:
+            self.status_label.setText("Ready")
+            return
+
+        self.status_label.setText("Transforming points ...")
+
+        self.run_transform_points_to_downsampled_space()
+        self.run_transform_downsampled_points_to_atlas_space()
+
+        self.status_label.setText("Analysing point distribution ...")
+        self.analyse_points()
+        self.status_label.setText("Ready")
+
+    def check_layers(self) -> bool:
+        """
+        Check if the layers needed to begin the transformation
+        have been selected by the user.
+
+        Returns
+        -------
+        bool
+            True if both raw data and points layers are selected,
+            False otherwise.
+        """
+        if self.raw_data is None and self.points_layer is None:
+            display_info(
+                self,
+                "No layers selected",
+                "Please select the layers corresponding to the points "
+                "you would like to transform and the raw data (registered by "
+                "brainreg)",
+            )
+            return False
+
+        if self.raw_data is None:
+            display_info(
+                self,
+                "No raw data layer selected",
+                "Please select a layer that corresponds to the raw "
+                "data (registered by brainreg)",
+            )
+            return False
+
+        if self.points_layer is None:
+            display_info(
+                self,
+                "No points layer selected",
+                "Please select a points layer you would like to transform",
+            )
+            return False
+
+        return True
+
+    def load_brainreg_directory(self) -> bool:
+        """
+        Load the brainreg directory selected by the user.
+        Returns false if not selected, to abort analysis.
+
+        Returns
+        -------
+        bool
+            True if a directory was selected, False otherwise.
+        """
+        brainreg_directory = QFileDialog.getExistingDirectory(
+            self,
+            "Select brainreg directory",
+        )
+        if brainreg_directory == "":
+            return False
+        else:
+            self.brainreg_directory = Path(brainreg_directory)
+
+        self.initialise_brainreg_data()
+        self.status_label.setText("Ready")
+        return True
+
+    def initialise_brainreg_data(self) -> None:
+        """
+        Initialize brainreg data by defining the paths,
+        then loading the brainreg metadata, and then the atlas.
+        """
+        self.get_brainreg_paths()
+        self.check_brainreg_directory()
+        self.get_registration_metadata()
+        self.load_atlas()
```

Comment:
```suggestion
        self.paths = Paths(self.brainreg_directory)
        self.check_brainreg_directory()
        self.metadata = Metadata(self.brainreg_metadata)
        self.load_atlas()
```

Would save some lines as we could remove two 1-line functions that are used only once, and avoid scrolling when reading the code to understand what's going on?

Reply:
Good idea, done.